### PR TITLE
Naming conventions

### PR DESCRIPTION
**Pull request by WoH**
- [x] Extract nested components, models and pipes.
- [x] Prefix selectors with poke-
- [x] Component Names according to conventions (and without the selector prefix)
- [x] import statements
- [x] Decorators
- [x] constructors
- [x] whitespace

angular.io/styleguide

### CODE REVIEW

**Comment by MajorBreakfast** [comment]
I'm pulling this now an see browse it through. Will respond in <10min

**Comment by MajorBreakfast** [comment]
- [x] We should make `selector` consistently the first property in `@Component`
- [ ] `<poke-filter>` has no selector
- [x] Selectors for `<map>`, `<nav-sidebar>`, `<navbar>` should be prefixed with `poke-`

**Comment by WoH** [comment]
I agree, was doing this while you were reviewing I guess :)
Just like the Decorator placement and the empty line between external and internal imports and the default constructors

**Comment by philbu** [comment]
Sorry for making merge conflicts. ;)

**Comment by MajorBreakfast** [comment]
Updated list above. Only selector for `<filter>` still missing. Good work!

**Comment by WoH** [comment]
@philbu np
@MajorBreakfast it won't need a selector and I think it could encourage someone to use it if i added one, that's why I didn't. Opinions?

**Comment by philbu** [comment]
It's not necessary but everyone who works actively on this project won't use it. You can decide.

**Comment by MajorBreakfast** [comment]
I'm for adding a selector to every component because of this:
named-selector

It's convenient for inspecting the page.

**Comment by MajorBreakfast** [comment]
I just checked all parts of the application and everything works.

You should add the missing selector, otherwise clear to merge. Good work!

File: ionic2/app/components/filter/filter.component.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { FilterTimeTabComponent } from '../filter-time-tab/filter-time-tab.compo
 import { FilterPokemonTabComponent } from '../filter-pokemon-tab/filter-pokemon-tab.component';
 
 @Component({
+  selector: 'poke-filter',
```
Comment (MajorBreakfast) [inline ionic2/app/components/filter/filter.component.ts:11]: 👍

**Comment by MajorBreakfast** [comment]
We should merge this quickly as it breaks pretty much any other possible PR :)

**Comment by WoH** [comment]
@philbu Do you agree with the refactor?

**Comment by philbu** [comment]
Yes, does look cleaner and more consistent

**Pull request merged**